### PR TITLE
mat: consolidate basic types for testing in single file

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

### CODE REVIEW

File: mat/basictypes_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import "gonum.org/v1/gonum/blas/blas64"
+
+func asBasicMatrix(d *Dense) *basicMatrix              { return (*basicMatrix)(d) }
+func asBasicVector(d *VecDense) *basicVector           { return (*basicVector)(d) }
+func asBasicSymmetric(s *SymDense) *basicSymmetric     { return (*basicSymmetric)(s) }
+func asBasicTriangular(t *TriDense) *basicTriangular   { return (*basicTriangular)(t) }
+func asBasicBanded(b *BandDense) *basicBanded          { return (*basicBanded)(b) }
+func asBasicSymBanded(s *SymBandDense) *basicSymBanded { return (*basicSymBanded)(s) }
+func asBasicTriBanded(t *TriBandDense) *basicTriBanded { return (*basicTriBanded)(t) }
+func asBasicDiagonal(d *DiagDense) *basicDiagonal      { return (*basicDiagonal)(d) }
```
Comment (kortschak) [inline mat/basictypes_test.go:16]: Are these all in a block at the top so we know what we have?

**Review by kortschak — APPROVED** [review]

File: mat/basictypes_test.go (hunk 1 above)
Comment (vladimir-ch) [reply mat/basictypes_test.go:16]: Possibly. I was split having a block here or have them with the type. I can change it to your preference.

Comment (kortschak) [reply mat/basictypes_test.go:16]: Either way is fine.

Comment (vladimir-ch) [reply mat/basictypes_test.go:16]: Then I'll leave it here, less work.

**Pull request merged**